JetNews/app/src/main/java/com/example/jetnews/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetnews.ui.home
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.viewModelScope
+import com.example.jetnews.R
+import com.example.jetnews.data.Result
+import com.example.jetnews.data.posts.PostsRepository
+import com.example.jetnews.data.succeeded
+import com.example.jetnews.model.Post
+import com.example.jetnews.utils.ErrorMessage
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+import java.util.UUID
+
+data class HomeUiState(
+    val posts: List<Post> = emptyList(),
+    val favorites: Set<String> = emptySet(),
```

Comment:
It looks like nothing is updating the favorites here, so toggling favorites doesn't update the UI.